code/clue.js

```diff
@@ -3,57 +3,274 @@
+  killerImage.src = mystery.killer.image;
+  killerAge.innerHTML = `${mystery.killer.age} years old`;
+  killerOccupation.innerHTML = `${mystery.killer.occupation}`;
+  killerDescription.innerHTML = `${mystery.killer.description}`;
+  killerFavouriteWeapon.innerHTML = `Favourite weapon: ${
+    randomFavouriteWeapon().name
+  }`;
+};
+//function to show weapon information
+const weaponInformation = () => {
+  const weaponName = document.getElementById('weaponName');
+  const weaponWeight = document.getElementById('weaponWeight');
+
+  mystery.weapon = randomSelector(weapons);
+  weaponLoader.style.display = 'none';
+  weaponHeadLine.innerHTML = 'Weapon-information';
+  weaponName.innerHTML = `Weapon: ${mystery.weapon.name}`;
+  weaponWeight.innerHTML = `Weapon weight: ${mystery.weapon.weight}`;
+};
+//function to show the weapon information
+const roomInformation = () => {
+  const roomName = document.getElementById('roomName');
+
+  mystery.room = randomSelector(rooms);
+  roomLoader.style.display = 'none';
+  roomHeadLine.innerHTML = 'Room:';
+  roomName.innerHTML = `${mystery.room}`;
+};
 
 // This function will be invoked when you click on the killer card.
 const pickKiller = () => {
-  // This will randomly select a killer from the suspects. And add that to the mystery object.
-  mystery.killer = randomSelector(suspects)
-
-  // This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer. Feel free to add more things to show about the killer.
-  document.getElementById('killerCard').style.background = mystery.killer.color
-  document.getElementById(
-    'killerName'
-  ).innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
-}
+  showKillerLoader();
+  setTimeout(killerInformation, 4000);
+};
 
 // CREATE FUNCTIONS pickWeapon and pickRoom in a similar way.
+const pickWeapon = () => {
+  showWeaponLoader();
+  setTimeout(weaponInformation, 4000);
+};
+
+const pickRoom = () => {
+  showRoomLoader();
+  setTimeout(roomInformation, 4000);
+};
+
+document.getElementById('killerCard').onclick = pickKiller;
+document.getElementById('weaponCard').onclick = pickWeapon;
+document.getElementById('roomCard').onclick = pickRoom;
+
+//The first that shows on the cards
+killerHeadLine.innerHTML = 'Pick a killer';
+weaponHeadLine.innerHTML = 'Pick a weapon';
+roomHeadLine.innerHTML = 'Pick a room';
 
 // STEP 4 - CREATE A FUNCTION revealMystery that will be invoked when you click that button. It should show something like:
 // 'The murder was committed by Jacob Green, in the living room with a rope.'
+const revealMystery = () => {
+  document.getElementById(
+    'mystery'
+  ).innerHTML = `The murder was commited by ${mystery.killer.firstName} ${mystery.killer.lastName} in the ${mystery.room} with a ${mystery.weapon.name}.`;
+};
+document.getElementById('revealButton').onclick = revealMystery;
```

Comment:
Not a requirement but could have been nice with some validation when I click the button :)

Reply:
Agreed, we should not have errors.
This happens when we don't click every card, but we try to reveal anyway.  Let's fix that @LillanEriksson 
![image](https://user-images.githubusercontent.com/807352/94700034-517c8180-033b-11eb-9568-0a6f8fc04ea8.png)

Reply:
Fixed it now!